hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/InputPathHandler.java

```diff
@@ -0,0 +1,138 @@
+  }
+
+  /**
+   * Takes in the original InputPaths and classifies each of them into incremental, non-incremental
+   * and non-hoodie InputPaths. The logic is as follows:
+   *
+   * 1. Check if an inputPath starts with the same basepath as any of the metadata basepaths we know
+   *    1a. If yes, this belongs to a Hoodie table that we already know about. Simply classify this
+   *        as incremental or non incremental - We can get the table name of this inputPath from the
+   *        metadata. Then based on the list of incrementalTables, we can classify this inputPath.
+   *    1b. If no, this could be a new Hoodie Table we haven't seen yet or a non-Hoodie Input Path.
+   *        Try creating the HoodieTableMetadataClient.
+   *            - If it succeeds, further classify as incremental on non-incremental as described in
+   *              step 1a above.
+   *            - If DatasetNotFoundException/InvalidDatasetException is caught, this is a
+   *              non-Hoodie inputPath
+   * @param inputPaths - InputPaths from the original jobConf that was passed to HoodieInputFormat
+   * @param incrementalTables - List of all incremental tables extracted from the config
+   * `hoodie.<table-name>.consume.mode=INCREMENTAL`
+   * @throws IOException
+   */
+  private void parseInputPaths(Path[] inputPaths, List<String> incrementalTables)
+      throws IOException {
+    for (int i = 0; i < inputPaths.length; i++) {
+      Path inputPath = inputPaths[i];
+      boolean basePathKnown = false;
+      for (HoodieTableMetaClient metaClient : tableMetaClientMap.values()) {
+        if (inputPath.toString().startsWith(metaClient.getBasePath())) {
+          // We already know this base path.
+          basePathKnown = true;
+          // Check if this is for a non incremental query
+          String tableName = metaClient.getTableConfig().getTableName();
+          tagAsIncrementalOrNonIncremental(inputPath, tableName, metaClient, incrementalTables);
+          break;
+        }
+      }
+      if (!basePathKnown) {
+        // This path is for a table that we dont know about yet.
+        HoodieTableMetaClient metaClient;
+        try {
+          metaClient = getTableMetaClient(inputPath.getFileSystem(conf), inputPath);
+          String tableName = metaClient.getTableConfig().getTableName();
+          tableMetaClientMap.put(tableName, metaClient);
+          tagAsIncrementalOrNonIncremental(inputPath, tableName, metaClient, incrementalTables);
+        } catch (DatasetNotFoundException | InvalidDatasetException e) {
+          // This is a non Hoodie inputPath
+          LOG.info("Handling a non-hoodie path " + inputPath);
+          nonHoodieInputPaths.add(inputPath);
+        }
+      }
+    }
+  }
+
+  private void tagAsIncrementalOrNonIncremental(Path inputPath, String tableName,
+      HoodieTableMetaClient metaClient, List<String> incrementalTables) {
+    if (incrementalTables == null || !incrementalTables.contains(tableName)) {
```

Comment:
ensuring we return a non-null list can simplifiy this if block? i.e just pass in an empty list if there are no incr tables?

Reply:
+1

Reply:
makes sense